middle_end/flambda/types/env/aliases.mli

```diff
@@ -31,9 +31,16 @@ val empty : t
 type add_result = private {
   t : t;
   canonical_element : Simple.t;
-  alias_of : Simple.t;
+  alias_of_demoted_element : Simple.t;
 }
 
+(** Add an alias relationship to the tracker. The two simple expressions
+    must be different and not both constants. If [add t s1 mode1 s2 mode2]
+    returns [{ t = t'; canonical_element; alias_of_demoted_element }], then
+    according to [t'],
+    - [canonical_element] is the canonical element of both [s1] and [s2];
+    - [alias_of_demoted_element] is either [s1] or [s2]; and
+    - [alias_of_demoted_element] is no longer canonical. *)
```

Comment:
Ah, good point, this is badly worded. I mean that, in the particular case that `s1` or `s2` was canonical to begin with, if `alias_of_demoted_element` is one of them, it's no longer canonical. Updating the comment. (This is important because `alias_of_demoted_element` is going to be given an alias type, which is invalid for a canonical element.)